src/runtime/backing.rs

```diff
@@ -0,0 +1,322 @@
+            to_init.copy_from_slice(&init.data);
+        }
+
+        memories
+            .iter_mut()
+            .map(|mem| mem.into_vm_memory())
+            .collect::<Vec<_>>()
+            .into_boxed_slice()
+    }
+
+    fn generate_tables(module: &Module) -> Box<[TableBacking]> {
+        let mut tables = Vec::with_capacity(module.tables.len());
+
+        for (_, table) in &module.tables {
+            let table_backing = TableBacking::new(table);
+            tables.push(table_backing);
+        }
+
+        tables.into_boxed_slice()
+    }
+
+    fn finalize_tables(
+        module: &Module,
+        imports: &ImportBacking,
+        tables: &mut [TableBacking],
+        sig_registry: &SigRegistry,
+    ) -> Box<[vm::LocalTable]> {
+        for init in &module.table_initializers {
+            assert!(init.base.is_none(), "global base not supported yet");
+            let table = &mut tables[init.table_index.index()];
+            match table.elements {
+                TableElements::Anyfunc(ref mut elements) => {
+                    for (i, &func_index) in init.elements.iter().enumerate() {
+                        let sig_index = module.signature_assoc[func_index];
+                        let vm_sig_id = sig_registry.get_vm_id(sig_index);
+
+                        let func_data = if module.is_imported_function(func_index) {
+                            imports.functions[func_index.index()].clone()
+                        } else {
+                            vm::ImportedFunc {
+                                func: (module.function_resolver)(module, func_index)
+                                    .unwrap()
+                                    .as_ptr(),
+                            }
+                        };
+
+                        elements[init.offset + i] = vm::Anyfunc {
+                            func_data,
+                            sig_id: vm_sig_id,
+                        };
+                    }
+                }
+            }
+        }
+
+        tables
+            .iter_mut()
+            .map(|table| table.into_vm_table())
+            .collect::<Vec<_>>()
+            .into_boxed_slice()
+    }
+
+    fn generate_globals(module: &Module) -> Box<[vm::LocalGlobal]> {
+        let globals = vec![vm::LocalGlobal::null(); module.globals.len()];
+
+        globals.into_boxed_slice()
+    }
+
+    fn finalize_globals(
+        module: &Module,
+        imports: &ImportBacking,
+        mut globals: Box<[vm::LocalGlobal]>,
+    ) -> Box<[vm::LocalGlobal]> {
+        for (to, (_, from)) in globals.iter_mut().zip(module.globals.into_iter()) {
+            to.data = match from.init {
+                GlobalInit::Val(Val::I32(x)) => x as u64,
+                GlobalInit::Val(Val::I64(x)) => x as u64,
+                GlobalInit::Val(Val::F32(x)) => x as u64,
+                GlobalInit::Val(Val::F64(x)) => x,
+                GlobalInit::GetGlobal(index) => (imports.globals[index.index()].global).data,
```

Comment:
I'm not sure I like having a verb (`Get`) into the struct name.
What you think about using `GlobalInit::ImportedGlobal` instead ?

Reply:
Okay, sounds fine.

Reply:
That struct is just exposing [instantiation-time initialization](https://github.com/sunfishcode/wasm-reference-manual/blob/master/WebAssembly.md#instantiation-time-initializers), which can either be a `const.*` or a `get_global`.

Reply:
Maybe *FromGlobal or sth. But that's the name Cranelift used as well.

Reply:
I'm just copying the name in the webassembly reference.